### PR TITLE
Hide hidden data parameters in workflow editor

### LINKED ISSUE #2050: bug with hidden_data?
https://github.com/galaxyproject/tools-devteam/issues/341
In release 16_01, devteam's cufflinks 2.2.1.0 was failed and I guess it's caused by a galaxy's bug with hidden_data. Actually I can't find how hidden_data is set.

In cufflinks_wrapper.xml, the following code 
...
`## Include reference annotation?`
`#if $reference_annotation.use_ref == "Use reference annotation":`
`-G $reference_annotation.reference_annotation_file`
`$reference_annotation.compatible_hits_norm`
`#end if`
`#if $reference_annotation.use_ref == "Use reference annotation guide":`
`-g $reference_annotation.reference_annotation_guide_file`
`--3-overhang-tolerance=$reference_annotation.three_overhang_tolerance`
`--intron-overhang-tolerance=$reference_annotation.intron_overhang_tolerance`
`$reference_annotation.no_faux_reads`
`#end if`
`...`
`## Include global model if available.`
`#if $global_model:`
`--global_model=$global_model`
`#end if`
`...`
`<param name="global_model" type="hidden_data" label="Global model (for use in Trackster)" optional="True"/>`

generated the following in tool_script.sh where `$global_model` is set to the value of `$reference_annotation.reference_annotation_guide_file`.

`python /home/galaxy/dev/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/64698e16f4a6/cufflinks/cufflinks_wrapper.py --input=/home/galaxy/dev/database/files/000/dataset_634.dat --assembled-isoforms-output=/home/galaxy/dev/database/files/000/dataset_654.dat --num-threads="${GALAXY_SLOTS:-4}" -I 300000 -F 0.1 -j 0.15   -g /home/galaxy/dev/database/files/000/dataset_26.dat --3-overhang-tolerance=600 --intron-overhang-tolerance=50   -b --ref_file=/data/hg19/hg19dip.fa  -u  --global_model=/home/galaxy/dev/database/files/000/dataset_26.dat`

### PR DESCRIPTION
**Pull request by guerler**
Fixes #2050.

### CODE REVIEW

File: lib/galaxy/workflow/modules.py [hunk 1]
```diff
@@ -904,22 +904,23 @@ def get_data_inputs( self ):
...
-                ) )
+            if hasattr( input, 'hidden' ) and not input.hidden:
+                if isinstance( input, DataToolParameter ):
+                    data_inputs.append( dict(
+                        name=prefixed_name,
+                        label=prefixed_label,
+                        multiple=input.multiple,
+                        extensions=input.extensions,
+                        input_type="dataset", ) )
+                if isinstance( input, DataCollectionToolParameter ):
```
Comment (nsoranzo) [inline lib/galaxy/workflow/modules.py]: s/if/elif/

just to make it clear.

File: lib/galaxy/workflow/modules.py [hunk 2]
```diff
@@ -904,22 +904,23 @@ def get_data_inputs( self ):
...
-            if isinstance( input, DataCollectionToolParameter ):
-                data_inputs.append( dict(
-                    name=prefixed_name,
-                    label=prefixed_label,
-                    multiple=input.multiple,
-                    input_type="dataset_collection",
-                    collection_types=input.collection_types,
-                    extensions=input.extensions,
-                ) )
+            if hasattr( input, 'hidden' ) and not input.hidden:
```
Comment (nsoranzo) [inline lib/galaxy/workflow/modules.py]: Not sure this is correct, I think this should be:

``` python
if not hasattr( input, 'hidden' ) or not input.hidden:
```

**Comment by guerler** [comment]
Thanks for the suggestions @nsoranzo.

**Comment by nsoranzo** [comment]
:+1:

**Pull request merged**